### PR TITLE
list_network() and Show_node for network

### PR DESCRIPTION
**Pull request by xuhang57**
Changed the api.py and cli.py for creating a new call: list_network() and have a unit test under tests/unit/api.py 

Also testing Ian's pull request for #469. Show_node() passes the unit test cases. 

One concern here.  "deferred.apply_networking()" is used in the script in api.py unit test. I know it is essential, but I don't know why. When I used the cli to test the show_node(), I got the error saying:

sqlalchemy.exc.UnboundExecutionError: Could not locate a bind configured on mapper Mapper|User|User or this Session

And I cannot figure it out so far.

### CODE REVIEW

File: haas/api.py [hunk 1]
```diff
@@ -116,14 +116,10 @@ def project_connect_node(project, node):
...
 
     If the node or project does not exist, a NotFoundError will be raised.
-
-    If node is already owned by a project, a BlockedError will be raised.
     """
     db = model.Session()
     project = _must_find(db, model.Project, project)
     node = _must_find(db, model.Node, node)
-    if node.project is not None:
-        raise BlockedError("Node is already owned by a project.")
```
Comment (henn) [inline haas/api.py]: Is this a merge error? We don't want to remove this check

File: haas/api.py [hunk 2]
```diff
@@ -534,6 +530,16 @@ def headnode_detach_network(headnode, hnic):
...
+@rest_call('GET', '/networks')
+def list_networks():
+    """List all networks.
+    Returns a JSON array of strings representing a list of projects.
+    Example:  '["network1", "network2", "network3"]'
+    """
+    db = model.Session()
+    networks = db.query(model.Network).all()
+    networks = [n.label for n in networks]
+    return json.dumps(networks, sort_keys=True)
```
Comment (henn) [inline haas/api.py]: Would it be possible to add an optional "verbose" here, which would also show you the vlan ID for each network (which is shown by `show_network()`)?

File: tests/unit/api.py [hunk 3]
```diff
@@ -235,17 +236,6 @@ def test_project_connect_node_node_nexist(self, db):
...
-        """Connecting a node which is not free to a project should fail."""
-        api.node_register('node-99', 'ipmihost', 'root', 'tapeworm')
-
-        api.project_create('anvil-oldtimer')
-        api.project_create('anvil-nextgen')
-
-        api.project_connect_node('anvil-oldtimer', 'node-99')
-        with pytest.raises(api.BlockedError):
-            api.project_connect_node('anvil-nextgen', 'node-99')
-
```
Comment (henn) [inline tests/unit/api.py]: merge error?

File: tests/unit/api.py [hunk 4]
```diff
@@ -1224,23 +1234,22 @@ def test_show_node(self, db):
...
             ],
         }
         self._compare_node_dumps(actual, expected)
 
-
-    def test_show_node_unavailable(self, db):
-        api.node_register('robocop', 'ipmihost', 'root', 'tapeworm')
-        api.node_register_nic('robocop', 'eth0', 'DE:AD:BE:EF:20:14')
-        api.node_register_nic('robocop', 'wlan0', 'DE:AD:BE:EF:20:15')
-
```
Comment (henn) [inline tests/unit/api.py]: merge error

**Comment by henn** [comment]
Thanks for your first PR, Lucas!

Please address the comments so we can get these changes merged in.

**Comment by xuhang57** [comment]
Getting a new PR due to the merge error

File: haas/api.py (hunk 2 above)
Comment (xuhang57) [inline haas/api.py]: I will see what I do with it!

**Pull request closed without merging**